classes/task/failed_task.php

```diff
@@ -31,6 +31,98 @@ public function get_name() {
         return get_string('taskfailed', 'logstore_xapi');
     }
 
+    private function extract_failed_events($limitnum) {
+        global $DB;
+        $conditions = null;
+        $sort = '';
+        $fields = '*';
+        $limitfrom = 0;
+        $extractedevents = $DB->get_records('logstore_xapi_failed_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
```

Comment:
if I understand well this one similar with:
  $extractedevents = $DB->get_records('logstore_xapi_failed_log', null, null, null, null, $limitnum);

We definitely don't need any extra declaration here because all of them are set in the function declaration.

Reply:
I think the original programmer wanted to be able to change this easily if necessary, so the parameters are declared explicitly.

Reply:
make sense.